Allow compressed external OVAL files as well as variables in check-content-ref

redhatrises opened the pull request:
- Fixes #2

canb227 commented:
Following the conversation on 4/16:

Add optional hash field.
Add plain text recommendations to check signatures on content.
Add supported formats list (Gabe already has this, add uncompressed XML format).

DavidRies commented:
Question: the table in this document and, consequently, the "rich" markup of this change here on this site appears to be broken/garbled making it difficult to review this change. Anyone else having this problem?

When I:
1. Click Files Changed Above
2. Click the [Document] icon to show the "rich text" diff (i.e. markdown)

I get a garble of pipes and brackets. Anyone have a tip for reviewing this change?

-David

solind commented:
Hey guys, regarding the list of MIME types, I have a few thoughts:

1) The ZIP and TAR are multi-file archive formats.  But an HREF can only refer to one file.  So, what are we trying to accomplish here?  If archives are permitted IFF they include exactly one file, that should be documented.
2) There is no need to add an optional checksum element.  If you knew the checksum of the file you're pointing to, when you might as well include the file content in the datastream.  Every time I've seen an href to a server-hosted file, it's been for something that changes on a schedule that's different (more frequent) vs the benchmark.  So I don't think anyone will necessarily know the checksum _a priori_.
3) We don't need to actually specify a MIME type, we could just list formats that MUST be supported, e.g., GZIP, BZ2, raw XML.  That's because the first 4 bytes or so streaming back from the HTTP GET will tell you what the file type is.

redhatrises commented:
> Following the conversation on 4/16:
> 
> Add optional hash field.
> Add plain text recommendations to check signatures on content.
> Add supported formats list (Gabe already has this, add uncompressed XML format).

Done and added.

DavidRies commented:
I am having trouble reviewing this because the tables are broken in GitHub (I just seem a garble of pipes). How are people reviewing this change?

djhaynes commented:
@DavidRies, when I look at the changes via the "Files Changed" tab, it looks properly formatted to me using the "source diff" option (<> icon). The "rich diff" option (document icon) gives me the garbled pipes.

DavidRies commented on docs/6_xccdf_model.md:
```diff
@@ -2126,6 +2150,17 @@ Table 11 shows the possible properties of the *\<xccdf:check\>* element.
 |                 |                 |                 | be resolved to  |
 |                 |                 |                 | provide         |
 |                 |                 |                 | content.        |
+|                 |                 |                 |                 |
```
I think we should discourage, remove or _at least_ strongly qualify the use of a checksum. Typically, in current use, external check-content is referenced for the express purpose of using content that is updated more frequently than the Benchmark. For example, Benchmark authors often reference a CVE OVAL feed rather than including the OVAL. If a content author were to add a checksum to a CVE feed reference, it would fail as soon as the CVE feed was updated, rendering the Rule uncheckable without ignoring the checksum.

On the other hand, if the rule author is referencing check content that is not updated more frequently than the Benchmark, then there is no need to reference it. It can be included in the Datastream.

DavidRies commented on docs/6_xccdf_model.md:
```diff
@@ -2322,6 +2357,146 @@ Table 11 shows the possible properties of the *\<xccdf:check\>* element.
+|                 |                 |                 |     extension   |
+|                 |                 |                 |     types       |
++-----------------+-----------------+-----------------+-----------------+
+| checksum        | string          | 0-1             | An optional str |
```
See comment above re checksum.

DavidRies commented on docs/6_xccdf_model.md:
```diff
@@ -2322,6 +2357,146 @@ Table 11 shows the possible properties of the *\<xccdf:check\>* element.
 |                 |                 |                 | Section         |
 |                 |                 |                 | 7.2.3.5.2.      |
 +-----------------+-----------------+-----------------+-----------------+
+| media-type      | string          | 0-1             | Applicable in   |
```
As per @solind, ZIP and TAR are multi-file archives. If we are going to support these, we'd need additional documentation for these. For simplicity sake, I think we should only support single-file archive types.

canb227 commented:
Flagged for NIST approval. Do not merge.

redhatrises commented:
Updated to use `markdown/xccdf.md` Removed checksum and added language that only a single file is supported vs multiple files that can exist in an archive.
